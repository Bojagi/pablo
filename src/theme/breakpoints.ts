export type Breakpoint = 'sm' | 'md' | 'lg' | 'xl';

export const breakpointNames: Breakpoint[] = ['sm', 'md', 'lg', 'xl'];

export type BreakpointsArray = [number, number, number, number];

export type Breakpoints = BreakpointsArray &
  Record<Breakpoint, number> & {
    breakpointNames: Breakpoint[];
  };

export function createBreakpoints(
  array: BreakpointsArray,
  usedBreakpointNames: Breakpoint[] = breakpointNames
): Breakpoints {
  const newBreakpoints = ([...array] as unknown) as Breakpoints;
  usedBreakpointNames.forEach((bp, index) => {
    newBreakpoints[bp] = newBreakpoints[index];
  });
  newBreakpoints.breakpointNames = usedBreakpointNames;

  return newBreakpoints as Breakpoints;
}

export const defaultBreakpoints: BreakpointsArray = [600, 1000, 1200, 1920];

export const breakpoints = createBreakpoints(defaultBreakpoints, breakpointNames);
