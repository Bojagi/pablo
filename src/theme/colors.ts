export type ShadedColor = {
  main: string;
  dark: string;
  darkest: string;
  light: string;
  lightest: string;
  contrastText: string;
  contrastTextLight: string;
};

export type CommonColors = {
  white: string;
  black: string;
  whiteContrastText: string;
  blackContrastText: string;
};

export type TextColors = {
  main: string;
  info: string;
};

export type BorderColors = {
  main: string;
  lightest: string;
};

export interface GrayColors {
  '50': string;
  '100': string;
  '200': string;
  '300': string;
  '400': string;
  '500': string;
  '600': string;
  '700': string;
  '800': string;
  '900': string;
}

export type AllColors = ShadedColor & TextColors & BorderColors & CommonColors;

export interface Colors {
  common: CommonColors;
  gray: GrayColors;
  borders: BorderColors;
  brand: ShadedColor;
  positive: ShadedColor;
  negative: ShadedColor;
  neutral: ShadedColor;
  text: TextColors;
}

export const WHITE = '#ffffff';
export const BLACK = '#000000';

export const colors: Colors = {
  common: {
    white: WHITE,
    black: BLACK,
    whiteContrastText: BLACK,
    blackContrastText: WHITE,
  },
  gray: {
    '50': getGrayColor(0.95),
    '100': getGrayColor(0.9),
    '200': getGrayColor(0.8),
    '300': getGrayColor(0.7),
    '400': getGrayColor(0.6),
    '500': getGrayColor(0.5),
    '600': getGrayColor(0.4),
    '700': getGrayColor(0.3),
    '800': getGrayColor(0.2),
    '900': getGrayColor(0.1),
  },
  borders: {
    main: `${BLACK}${getHexByte(0.25)}`,
    lightest: `${BLACK}${getHexByte(0.1)}`,
  },
  text: {
    main: BLACK,
    info: `${BLACK}${getHexByte(0.5)}`,
  },
  brand: {
    lightest: '#EBE7FF',
    light: '#D6CEFF',
    main: '#6A50F2',
    dark: '#4129BD',
    darkest: '#281683',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  positive: {
    lightest: '#DFF6E6',
    light: '#BBECCB',
    main: '#38C765',
    dark: '#2FB85A',
    darkest: '#108B36',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  negative: {
    lightest: '#FFE7E5',
    light: '#FFCFCC',
    main: '#F21D0D',
    dark: '#CA271B',
    darkest: '#871A12',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  neutral: {
    lightest: '#FCF5DB',
    light: '#FFEB9E',
    main: '#FFDD5B',
    dark: '#EDC52A',
    darkest: '#D6AC0B',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
};

/**
 * Get hex color byte (e.g. to get transparency)
 * @param value Percentage value (between 0.0 and 1.0)
 */
function getHexByte(value: number) {
  return (Math.round((255 * value) / 16) * 16).toString(16);
}

function getGrayColor(value: number) {
  return `#${getHexByte(value)}${getHexByte(value)}${getHexByte(value)}`;
}
