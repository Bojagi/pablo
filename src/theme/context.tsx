import * as React from 'react';
import { PabloTheme, ComponentStyles } from './types';

export const pabloThemeContext = React.createContext<PabloTheme>({} as any);
export const pabloComponentStylesContext = React.createContext<ComponentStyles>({} as any);
export const rootContext = React.createContext<Document | ShadowRoot | null>(null);

export const useComponentStyleContext = () => React.useContext(pabloComponentStylesContext);

export const usePabloTheme = () => React.useContext(pabloThemeContext);
