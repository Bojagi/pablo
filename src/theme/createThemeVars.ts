import isObject from 'isobject';

type DeepStringReplace<T, M extends [any, any]> = {
  [P in keyof T]: T[P] extends object ? DeepStringReplace<T[P], M> : string;
};

export function createThemeVars<T extends Record<string, any>>(base: string, themeObject: T) {
  return internalThemeVars(`pbl-theme-${base}`, themeObject);
}

function internalThemeVars<T extends Record<string, any>>(
  base: string,
  themeObject: T
): DeepStringReplace<T, [number, string]> {
  return Object.entries(themeObject).reduce((acc, [key, value]) => {
    const keyVar = `${base}-${key}`;
    return { ...acc, [key]: isObject(value) ? internalThemeVars(keyVar, value) : `${keyVar}` };
  }, {} as any);
}

export function createThemeVarArray<T extends readonly string[]>(
  base: string,
  values: T
): Record<T[number], string> {
  return values.reduce(
    (acc, value) => ({ ...acc, [value]: `pbl-theme-${base}-${value}` }),
    {} as any
  );
}
