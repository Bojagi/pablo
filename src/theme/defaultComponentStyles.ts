import { buttonStyles, ButtonStyles } from '../Button/styles';
import { CardStyles, cardStyles } from '../Card/styles';
import { sidebarNavStyles, SidebarNavStyles } from '../SidebarNav/styles';
import { SwitchStyles, switchStyles } from '../Switch/styles';
import { TabsStyles, tabsStyles } from '../Tabs/styles';

export interface ComponentStyles {
  card: CardStyles;
  tabs: TabsStyles;
  sidebarNav: SidebarNavStyles;
  button: ButtonStyles;
  switch: SwitchStyles;
}

export const defaultComponentStyles: ComponentStyles = {
  card: cardStyles,
  tabs: tabsStyles,
  sidebarNav: sidebarNavStyles,
  button: buttonStyles,
  switch: switchStyles,
};
