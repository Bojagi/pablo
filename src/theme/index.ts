export * from './defaultTheme';
export * from './context';
export * from './PabloThemeProvider';
export * from './useComponentStyle';
