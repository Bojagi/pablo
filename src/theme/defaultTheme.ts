import { spacingTokens } from './spacing';
import { PabloTheme } from './types';
import { colors } from './colors';
import { typography } from './typography';
import { fluid } from './fluid';
import { breakpoints } from './breakpoints';

export const defaultTheme: PabloTheme = {
  spacing: spacingTokens,
  colors,
  typography,
  breakpoints,
  fluid,
};
