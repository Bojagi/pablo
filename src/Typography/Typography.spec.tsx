import { render } from '@testing-library/react';
import React from 'react';
import { PabloThemeProvider } from '../theme';
import {
  ButtonTypography,
  Headline,
  InfoText,
  InfoTextBold,
  Paragraph,
  ParagraphBold,
  Subtitle,
  Title,
  Typography,
} from './Typography';

describe('Typography component', () => {
  test.each([
    [undefined],
    ['headline'],
    ['title'],
    ['subtitle'],
    ['paragraph'],
    ['paragraphBold'],
    ['button'],
    ['info'],
    ['infoBold'],
  ])('styles with "%s" variant', (variant) => {
    const { container } = renderComponent({ variant });
    expect(container).toMatchSnapshot();
  });
});

describe('specific typography components', () => {
  test.each([
    ['Headline', Headline],
    ['Title', Title],
    ['Subtitle', Subtitle],
    ['Paragraph', Paragraph],
    ['ParagraphBold', ParagraphBold],
    ['ButtonTypography', ButtonTypography],
    ['InfoText', InfoText],
    ['InfoBold', InfoTextBold],
  ])('styles of "%s" component', (_stringName, Component) => {
    const { container } = render(
      <PabloThemeProvider>
        <Component>Hello</Component>
      </PabloThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });
});

function renderComponent(props) {
  return render(
    <PabloThemeProvider>
      <Typography {...props}>Hello</Typography>
    </PabloThemeProvider>
  );
}
