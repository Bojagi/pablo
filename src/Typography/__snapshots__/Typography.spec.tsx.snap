// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline component  Render component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h2-lineHeight);
  font-size: var(--pbl-theme-typography-h2-fontSize);
  margin-bottom: var(--pbl-theme-typography-h2-marginBottom);
  font-weight: var(--pbl-theme-typography-h2-fontWeight);
}

<div>
  <h2
    class="emotion-0"
  >
    Hello
  </h2>
</div>
`;

exports[`InfoText component  Render component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Paragraph component  Render component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`ParagraphBold component  Render component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Subtitle component  Render component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h4-lineHeight);
  font-size: var(--pbl-theme-typography-h4-fontSize);
  margin-bottom: var(--pbl-theme-typography-h4-marginBottom);
  font-weight: var(--pbl-theme-typography-h4-fontWeight);
}

<div>
  <h4
    class="emotion-0"
  >
    Hello
  </h4>
</div>
`;

exports[`Title component  Render component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h3-lineHeight);
  font-size: var(--pbl-theme-typography-h3-fontSize);
  font-weight: var(--pbl-theme-typography-h3-fontWeight);
  margin-bottom: var(--pbl-theme-typography-h3-marginBottom);
}

<div>
  <h3
    class="emotion-0"
  >
    Hello
  </h3>
</div>
`;

exports[`Typography component with variant body Render component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant body Render component 2`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant body Render component 3`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant body Render component 4`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant body Render inline component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant body Render inline component 2`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant body Render inline component 3`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant body Render inline component 4`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant button Render component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-button-lineHeight);
  font-size: var(--pbl-theme-typography-button-fontSize);
  margin-bottom: var(--pbl-theme-typography-button-marginBottom);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-button-variants-small-fontSize);
  font-weight: var(--pbl-theme-typography-button-variants-small-fontWeight);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant button Render component 2`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-button-lineHeight);
  font-size: var(--pbl-theme-typography-button-fontSize);
  margin-bottom: var(--pbl-theme-typography-button-marginBottom);
  font-size: var(--pbl-theme-typography-button-variants-small-fontSize);
  font-weight: var(--pbl-theme-typography-button-variants-small-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-button-variants-small-fontSize);
  font-weight: var(--pbl-theme-typography-button-variants-small-fontWeight);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant button Render inline component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-button-lineHeight);
  font-size: var(--pbl-theme-typography-button-fontSize);
  margin-bottom: var(--pbl-theme-typography-button-marginBottom);
  margin-bottom: 0;
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-button-variants-small-fontSize);
  font-weight: var(--pbl-theme-typography-button-variants-small-fontWeight);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant button Render inline component 2`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-button-lineHeight);
  font-size: var(--pbl-theme-typography-button-fontSize);
  margin-bottom: var(--pbl-theme-typography-button-marginBottom);
  margin-bottom: 0;
  font-size: var(--pbl-theme-typography-button-variants-small-fontSize);
  font-weight: var(--pbl-theme-typography-button-variants-small-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-button-variants-small-fontSize);
  font-weight: var(--pbl-theme-typography-button-variants-small-fontWeight);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant h1 Render component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h1-lineHeight);
  font-size: var(--pbl-theme-typography-h1-fontSize);
  margin-bottom: var(--pbl-theme-typography-h1-marginBottom);
  font-weight: var(--pbl-theme-typography-h1-fontWeight);
}

<div>
  <h1
    class="emotion-0"
  >
    Hello
  </h1>
</div>
`;

exports[`Typography component with variant h1 Render inline component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h1-lineHeight);
  font-size: var(--pbl-theme-typography-h1-fontSize);
  margin-bottom: var(--pbl-theme-typography-h1-marginBottom);
  font-weight: var(--pbl-theme-typography-h1-fontWeight);
  margin-bottom: 0;
}

<div>
  <h1
    class="emotion-0"
  >
    Hello
  </h1>
</div>
`;

exports[`Typography component with variant h2 Render component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h2-lineHeight);
  font-size: var(--pbl-theme-typography-h2-fontSize);
  margin-bottom: var(--pbl-theme-typography-h2-marginBottom);
  font-weight: var(--pbl-theme-typography-h2-fontWeight);
}

<div>
  <h2
    class="emotion-0"
  >
    Hello
  </h2>
</div>
`;

exports[`Typography component with variant h2 Render inline component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h2-lineHeight);
  font-size: var(--pbl-theme-typography-h2-fontSize);
  margin-bottom: var(--pbl-theme-typography-h2-marginBottom);
  font-weight: var(--pbl-theme-typography-h2-fontWeight);
  margin-bottom: 0;
}

<div>
  <h2
    class="emotion-0"
  >
    Hello
  </h2>
</div>
`;

exports[`Typography component with variant h3 Render component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h3-lineHeight);
  font-size: var(--pbl-theme-typography-h3-fontSize);
  font-weight: var(--pbl-theme-typography-h3-fontWeight);
  margin-bottom: var(--pbl-theme-typography-h3-marginBottom);
}

<div>
  <h3
    class="emotion-0"
  >
    Hello
  </h3>
</div>
`;

exports[`Typography component with variant h3 Render inline component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h3-lineHeight);
  font-size: var(--pbl-theme-typography-h3-fontSize);
  font-weight: var(--pbl-theme-typography-h3-fontWeight);
  margin-bottom: var(--pbl-theme-typography-h3-marginBottom);
  margin-bottom: 0;
}

<div>
  <h3
    class="emotion-0"
  >
    Hello
  </h3>
</div>
`;

exports[`Typography component with variant h4 Render component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h4-lineHeight);
  font-size: var(--pbl-theme-typography-h4-fontSize);
  margin-bottom: var(--pbl-theme-typography-h4-marginBottom);
  font-weight: var(--pbl-theme-typography-h4-fontWeight);
}

<div>
  <h4
    class="emotion-0"
  >
    Hello
  </h4>
</div>
`;

exports[`Typography component with variant h4 Render inline component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h4-lineHeight);
  font-size: var(--pbl-theme-typography-h4-fontSize);
  margin-bottom: var(--pbl-theme-typography-h4-marginBottom);
  font-weight: var(--pbl-theme-typography-h4-fontWeight);
  margin-bottom: 0;
}

<div>
  <h4
    class="emotion-0"
  >
    Hello
  </h4>
</div>
`;

exports[`Typography component with variant undefined Render component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant undefined Render component 2`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant undefined Render component 3`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant undefined Render inline component 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant undefined Render inline component 2`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;

exports[`Typography component with variant undefined Render inline component 3`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

.emotion-0 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-0 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <p
    class="emotion-0"
  >
    Hello
  </p>
</div>
`;
